language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/StartActionNodeContext.java

```diff
@@ -0,0 +1,102 @@
+import io.ballerina.compiler.syntax.tree.StartActionNode;
+import io.ballerina.tools.text.TextRange;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.common.utils.completion.QNameReferenceUtil;
+import org.ballerinalang.langserver.commons.BallerinaCompletionContext;
+import org.ballerinalang.langserver.commons.completion.LSCompletionException;
+import org.ballerinalang.langserver.commons.completion.LSCompletionItem;
+import org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.ballerina.compiler.api.symbols.SymbolKind.CLASS;
+import static io.ballerina.compiler.api.symbols.SymbolKind.FUNCTION;
+
+/**
+ * Handles the completions for {@link StartActionNode} context.
+ *
+ * @since 2.0.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.completion.spi.BallerinaCompletionProvider")
+public class StartActionNodeContext extends AbstractCompletionProvider<StartActionNode> {
+
+    public StartActionNodeContext() {
+        super(StartActionNode.class);
+    }
+
+    @Override
+    public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, StartActionNode node)
+            throws LSCompletionException {
+
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+
+        completionItems.addAll(this.expressionCompletions(context, node));
+        this.sort(context, node, completionItems);
+        return completionItems;
+    }
+
+    @Override
+    protected List<LSCompletionItem> expressionCompletions(BallerinaCompletionContext context,
+                                                           StartActionNode startActionNode) {
+
+        NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
+        if (QNameReferenceUtil.onQualifiedNameIdentifier(context, nodeAtCursor)) {
+            /*
+            Supports the following
+            (1) future<int> f = start module:<cursor>
+            (2) future<int> f = start module:a<cursor>
+             */
+            QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nodeAtCursor;
+            List<Symbol> ctxEntries = QNameReferenceUtil.getExpressionContextEntries(context, qNameRef);
+
+            return this.getCompletionItemList(ctxEntries, context);
+        }
+        /*
+        Supports the following
+        (1) future<int> f = start <cursor>
+        (2) future<int> f = start a<cursor>
+         */
+        List<LSCompletionItem> completionItems = new ArrayList<>(this.getTypeDescContextItems(context));
+        
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
+        List<Symbol> filteredList = visibleSymbols.stream()
+                .filter(symbol -> (symbol instanceof VariableSymbol || symbol.kind() == FUNCTION
+                        || symbol.kind() == CLASS))
```

Comment:
Why do we filter classes here? Class symbols are basically provided with `getTypeDescContextItems`, doing this will duplicate class entires.